model/task/db.go

```diff
@@ -812,6 +812,63 @@ func GetRecentTaskStats(period time.Duration, nameKey string) ([]StatusItem, err
 	return result, nil
 }
 
+// ByExecutionTasksAndMaxExecution returns the tasks corresponding to the passed in taskIds and execution,
+// or the most recent executions of those tasks if they do not have a matching execution
+func ByExecutionTasksAndMaxExecution(taskIds []*string, execution int) ([]Task, error) {
+	pipeline := []bson.M{}
+	match := bson.M{
+		"$match": bson.M{
+			IdKey: bson.M{
+				"$in": taskIds,
+			},
+			ExecutionKey: bson.M{
+				"$lte": execution,
+			},
+		},
+	}
+	pipeline = append(pipeline, match)
+	// Get the taskIds that were not found in the previous match stage
+	result := []Task{}
+	if err := Aggregate(pipeline, &result); err != nil {
+		return nil, errors.Wrap(err, "Error finding tasks in task collection")
+	}
+	taskMap := map[string]bool{}
+	for _, t := range result {
+		taskMap[t.Id] = true
+	}
+	missingTasks := []string{}
```

Comment:
You can shorten this up by building a list of foundTaskIds and then call `missingTasks, _ := utility.StringSliceSymmetricDifference(taskIds, foundTaskIds)` because the first return value is all strings that exist in the first list (taskIds) but not in the second (found task Ids).